sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/KafkaConfiguration.java

```diff
@@ -25,8 +24,6 @@ public class KafkaConfiguration {
 
   private Map<String, ConsumerConfig> consumers = new HashMap<>();
 
-  private Map<String, TopicConfig> topics = new HashMap<>();
```

Comment:
I think we should provide one preferred way how we want users of the library to define/configure topic names. Usually they would need to override the specific topic name used in their environment. Options that come to my mind:

* we keep this key but only give the opportunity to set the topic name
* we use a convention that we try to lookup "KAFKA_TOPIC_NAME_<topic-name>" in the environment

What do you think @elbiocaetano / @JoergSiebahn / @eikenilsknopp / @preethivenkatesh-sdase ?

Reply:
What should `KAFKA_TOPIC_NAME_` refer to? So, what should we lookup at the end of the property?

Although it would cause refactoring in the services when upgrading, I would prefer not to keep a map just for a single value and not to do some unknown magic with environment variables.

I would suggest for the producer services, to add dedicated string properties in their `config.yml` / `AppConfiguration` or to extend the `KafkaConfiguration` to add their specific properties like `myModelTopicName`. The environment variables in the `config.yml` can be the same as before but write to a new property. This way, the service will be backward compatible to existing deployments. We have various services where we use inheritance for configuring Api clients. We also tried [to migrate that into the lib](https://github.com/SDA-SE/sda-dropwizard-commons/pulls?q=is%3Apr+is%3Aclosed+dedicated+api+client+configuration), but never finished it.

But to make the migration smoother, we could deprecate all properties but `name` in `TopicConfig` in v2 and keep this single property in v3 and think about a long term solution when v3 is released.